### PR TITLE
Adjust query function to get latest Runtimestate with Kyma version

### PR DESCRIPTION
**Pull request by szwedm**
**Description**

Changes proposed in this pull request:

- replace GetLatestRuntimeStatesByRuntimeID() with GetLatestRuntimeStateWithKymaVersionByRuntimeID(),
- adjust GetLatestWithKymaVersionByRuntimeID(),
- provide test case for fetching the latest Runtimestate with Kyma version by RuntimeId

**Related issue(s)**

Ref: #1264

### CODE REVIEW

File: components/kyma-environment-broker/internal/storage/postsql/read.go [hunk 1]
```diff
@@ -429,33 +429,29 @@ func (r readSession) GetLatestRuntimeStateWithReconcilerInputByRuntimeID(runtime
...
 		if err == dbr.ErrNotFound {
-			return states, dberr.NotFound("cannot find latest 10 runtime states: %s", err)
+			return state, dberr.NotFound("cannot find latest runtime state with kyma version: %s", err)
 		}
-		return states, dberr.Internal("Failed to get the latest 10 runtime states: %s", err)
+		return state, dberr.Internal("Failed to get the latest runtime states with kyma version: %s", err)
 	}
 	if count == 0 {
-		return states, dberr.NotFound("cannot find latest 10 runtime states with reconciler input: %s", err)
+		return state, dberr.NotFound("cannot find latest runtime states with kyma version: %s", err)
```
Comment (wozniakjan) [inline components/kyma-environment-broker/internal/storage/postsql/read.go:452]: nit: `states` => `state`. Or even better maybe, what would you say we rephrase my original but unfortunate wording to `found 0 latest runtime states with kyma version`, that would clearly differentiate this and error few lines higher when KEB deasl with `dbr.ErrNotFound`.

**Review by wozniakjan — DISMISSED** [review]
/lgtm
/hold

feel free to unhold if you find the nits below unnecessary

File: components/kyma-environment-broker/internal/storage/postsql/read.go [hunk 2]
```diff
@@ -429,33 +429,29 @@ func (r readSession) GetLatestRuntimeStateWithReconcilerInputByRuntimeID(runtime
...
-		Load(&states)
+		Limit(1).
+		Load(&state)
 	if err != nil {
 		if err == dbr.ErrNotFound {
-			return states, dberr.NotFound("cannot find latest 10 runtime states: %s", err)
+			return state, dberr.NotFound("cannot find latest runtime state with kyma version: %s", err)
 		}
-		return states, dberr.Internal("Failed to get the latest 10 runtime states: %s", err)
+		return state, dberr.Internal("Failed to get the latest runtime states with kyma version: %s", err)
```
[the remark this answers is not in the document]
Comment (szwedm) [reply components/kyma-environment-broker/internal/storage/postsql/read.go:449]: fixed

File: components/kyma-environment-broker/internal/storage/postsql/read.go (hunk 1 above)
Comment (szwedm) [reply components/kyma-environment-broker/internal/storage/postsql/read.go:452]: good point, I changed the error msg as you suggested

**Review by wozniakjan — APPROVED** [review]
/lgtm
/unhold

**Pull request merged**